### PR TITLE
[New Hack] - 062 Entra ID B2B

### PR DESCRIPTION
**Pull request by bappadityams**
Entra ID WTH challenges and solutions

### CODE REVIEW

File: 062-EntraIDB2B/Images/depoy-aad.png
Comment (perktime) [inline 062-EntraIDB2B/Images/depoy-aad.png:1]: Note that this diagram says Azure AD and not Entra ID

File: 062-EntraIDB2B/Images/spa-angular.png
Comment (perktime) [inline 062-EntraIDB2B/Images/spa-angular.png:1]: Note that this diagram says Azure AD not Entra ID

**Review by perktime — APPROVED** [review]

File: 062-EntraIDB2B/Student/Challenge-02.md
Comment (Whowong) [inline 062-EntraIDB2B/Student/Challenge-02.md:1]: This challenge looks light.  Does it make sense to combine with challenge 1 as part of the requirements to also login?

File: 062-EntraIDB2B/Student/Challenge-03.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+## Description
+
+With Entra ID B2B, the partner uses their own identity management solution, so there's no external administrative overhead for your organization. Guest users sign in to your apps and services with their own work, school, or social identities.
+
+- The partner uses their own identities and credentials, whether or not they have an Entra ID account.
+- You don't need to manage external accounts or passwords.
+- You don't need to sync accounts or manage account lifecycle.
+
+When sharing an application with external users, you might not always know in advance who will need access to the application. As an alternative to sending invitations directly to individuals, you can allow external users to sign up for specific applications themselves by enabling self-service sign-up user flow. You can create a personalized sign-up experience by customizing the self-service sign-up user flow. For example, you can provide options to sign up with Entra ID or social identity providers and collect information about the user during the sign-up process.
+
```
Comment (Whowong) [inline 062-EntraIDB2B/Student/Challenge-03.md:21]: This was a great way to understand the benefits and use cases of Entra.  However, it was difficult to understand what we wanted the students to accomplish without looking at the success criteria.  Can we add/clarify the description?

File: 062-EntraIDB2B/Student/Challenge-06.md
Comment (Whowong) [inline 062-EntraIDB2B/Student/Challenge-06.md:1]: Are we deploying the app that we built in challenge 5?  If so, we should limit the different applications you provided as I saw some were desktop applications in challenge 5

File: 062-EntraIDB2B/Student/Challenge-05.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Challenge 05 - Integrate your application with Entra ID
+
+[< Previous Challenge](./Challenge-04.md) - **[Home](../README.md)** - [Next Challenge >](./Challenge-06.md)
+
+
+## Introduction
+
+The code samples are built and maintained by Microsoft to demonstrate usage of our authentication libraries with the Microsoft identity platform. Common authentication and authorization scenarios are implemented in several application types, development languages, and frameworks.
+
+Each tutorial / walkthrough describes how to build the project (if applicable) and run the sample application. Comments in the code help you understand how these libraries are used in the application to perform authentication and authorization by using the identity platform.
```
Comment (Whowong) [inline 062-EntraIDB2B/Student/Challenge-05.md:10]: The walkthrough feel more prescriptive than what a WTH should be.  Is there a way to challenge the students more?

File: 062-EntraIDB2B/Student/Challenge-02.md
Comment (bappadityams) [reply 062-EntraIDB2B/Student/Challenge-02.md:1]: We would like to make the students understand the sign in process separately, oAuth login, passing Client ID, tenant ID etc. And there is a little bit of set up needed as well(user creation, authr client set up etc). As part of the challenge, student can explore different sign in options as well.

File: 062-EntraIDB2B/Student/Challenge-03.md (hunk 1 above)
Comment (bappadityams) [reply 062-EntraIDB2B/Student/Challenge-03.md:21]: Updated the description.

File: 062-EntraIDB2B/Student/Challenge-06.md
Comment (bappadityams) [reply 062-EntraIDB2B/Student/Challenge-06.md:1]: Yes, we will be deploying the app built in challenge 5. I updated the description of Challenge 6 to make it more clear and limit the with .Net/SPA.

**Review by perktime — APPROVED** [review]

**Review by perktime — APPROVED** [review]

**Review by perktime — APPROVED** [review]

**Review by Whowong — APPROVED** [review]

**Pull request merged**